### ISSUE TITLE
Player stat calculation is wrong when loading game while in combat

### ISSUE THREAD

**Issue by Nelson-Lai** [MelvorIdle/melvoridle.github.io#1214]
**Describe the bug**
TLDR: I never die when I idle online combat, but on offline combat I die.

I am trying offline combat grinding Shockwave Fragments from Rokken. I have idled the game in a tab for "Online" combat successfully for over 12 hours, but when I try offline combat it will say I die rather quickly. Perhaps I am not auto-eating correctly in offline progression or DR is not applied correctly.

**To Reproduce**
Sorry reproducibility is difficult because of combat RNG rolls, I think that I might be losing some DR on my summon or something else. 

My equipment:
![image](https://user-images.githubusercontent.com/50011161/129466079-a064e74b-41ab-4e35-a8b6-71803f94308c.png)

Combat stats:
![image](https://user-images.githubusercontent.com/50011161/129466084-dbb7d052-a29d-4a18-8afa-1532ad979763.png)

Max hit:
![image](https://user-images.githubusercontent.com/50011161/129466088-0603e02e-1484-4d8d-8909-9cd4f554c8ae.png)
Note this is while stunned:

Max hit: (Not stunned)
![image](https://user-images.githubusercontent.com/50011161/129466098-e3ff08cf-a6f9-4df8-8e7b-91060308b459.png)

Auto Eat:
![image](https://user-images.githubusercontent.com/50011161/129466102-d7cf3157-d869-4558-a7f2-cc1ee01d14e6.png)

**Expected behavior**
As the Rokken max hit at my level of DR (67% after bonus) is 407 (Shockwave while stunned from previous shockwave), and my auto eat is 444 or below, I expect to never die if I have food available.

**Screenshots**
Sorry, the screenshot is just "You Died!". I didnt take the screenshot but I can replicate if necessary

**Browser**
Chrome 

**Console output**
Not sure this is relevant, what parts of the console are you looking to see?

**Are you using any scripts?**
No, base web version

**Comment by Randomuser1730** [comment]
I've noticed the same issue recently on STEAM. Found someone ("Thijs1444") bring it up for Rokken as well in the Bug Discussion section for Discord and now you. I've been able to recreate it by fighting Rokken and then closing the game window. Loading it back up almost immediately (say before I've had a chance to kill Rokken once while offline) and upon reconnecting to the game I've lost the x1.25 Damage Modifier buff. As a result Rokkens damage increases considerably and when he Stuns me, depending on my health, his total damage is beyond what I can safely tank.

For me uneqipping/equipping an item or changing prayers immediately resets the values and I'm safe again - or simply surviving until the very next kill - where the DR Modifier is in place again. That said I have quit out in the middle of a fight where I still had my full DR in place and came back minutes later to death. I think for now the safest method of fighting Rokken is by running, closing the game - reloading - starting a fresh fight and then closing the game again, but I could be mistaken.

I've also tested this on other monsters and it seems to persist regardless of the enemy so long as they have a Damage Reduction Modifier in place for/against me.

**Comment by coolrox95** [comment]
Issue has been resolved as of Alpha v0.21 (?1112).
